open_schools_platform/organization_management/organizations/views.py

```diff
@@ -277,26 +276,26 @@ def get(self, request, pk):
         return Response({"analytics": AnalyticsSerializer(count_queries_by_statuses(queries)).data}, status=200)
 
 
-class QueriesCirclesOrganizationStudent(ApiAuthMixin, ListAPIView):
+class QueriesOrganizationStudent(ApiAuthMixin, ListAPIView):
     pagination_class = ApiCircleListPagination
-    serializer_class = CircleSerializer
+    serializer_class = StudentProfileQuerySerializer
 
     @swagger_auto_schema(
         tags=[SwaggerTags.ORGANIZATION_MANAGEMENT_ORGANIZATIONS],
         operation_description="Returns queries to student circles from the selected organization",
     )
-    def get(self, request, organization_id, student_profile_id, *args, **kwargs):
+    def get(self, request, organization, student_profile, *args, **kwargs):
         get_student_profile(
-            filters={"id": str(student_profile_id)},
+            filters={"id": str(student_profile)},
             empty_exception=True
         )
         organization = get_organization(
-            filters={"id": str(organization_id)},
+            filters={"id": str(organization)},
             user=request.user,
             empty_exception=True
         )
         queries = get_queries(
-            filters={"sender_id": str(student_profile_id),
+            filters={"sender_id": str(student_profile),
                      "recipient_ids": form_ids_string_from_queryset(organization.circles.values())}
         )
         return Response({"results": StudentProfileQuerySerializer(queries, many=True).data}, status=200)
```

Comment:
It is documented in swagger that the response is paginated, but your response is just an array of queries. Maybe you should use get_paginated_response function

Reply:
what do you mean? do "pagination_class = get_paginated_response()" or no?

Reply:
like [here](https://github.com/LamArt/open-schools-platform/blob/dev/open_schools_platform/organization_management/circles/views.py#L76)